src/assets/Azure.Core.Shared/JsonElementExtensions.cs

```diff
@@ -0,0 +1,46 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Text.Json;
+using System.Xml;
+
+namespace Azure.Core
+{
+    internal static class JsonElementExtensions
+    {
+        public static object? GetObject(in this JsonElement element)
+        {
+            switch (element.ValueKind)
+            {
+                case JsonValueKind.String:
+                    return element.GetString();
+                case JsonValueKind.Number:
+                    return element.GetInt32();
```

Comment:
Shouldn't we assume the largest amount of possible numeric types, and use decimal? (not int)

Reply:
Decimal seems overkill, I did double.